crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -298,30 +300,36 @@ impl ShaderCache {
     }
 }
 
+type LayoutCacheKey = (Vec<BindGroupLayoutId>, Vec<PushConstantRange>);
 #[derive(Default)]
 struct LayoutCache {
-    layouts: HashMap<Vec<BindGroupLayoutId>, ErasedPipelineLayout>,
+    layouts: PreHashMap<LayoutCacheKey, ErasedPipelineLayout>,
 }
 
 impl LayoutCache {
     fn get(
         &mut self,
         render_device: &RenderDevice,
         bind_group_layouts: &[BindGroupLayout],
+        push_constant_ranges: &[PushConstantRange],
```

Comment:
Maybe just accept a `Vec<PushConstantRange>` since that's ultimately what the implementation needs. Might be able to elide a `to_vec` or `clone` in some cases.